### PR TITLE
docs: updates to troubleshooting doc

### PR DESCRIPTION
**Pull request by Ulexus**
Updated wording, added some cross references, and generally updated the
controlplane troubleshooting document.

Signed-off-by: Seán C McCord <ulexus@gmail.com>

# Pull Request

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/5032)

### CODE REVIEW

File: website/content/docs/v0.15/Guides/troubleshooting-control-plane.md [hunk 1]
```diff
@@ -418,10 +474,12 @@ spec:
 
 Control plane nodes have enough secret material to generate `apid` server certificates, but worker nodes
 depend on control plane `trustd` services to generate certificates.
-Worker nodes wait for `kubelet` to join the cluster, then `apid` queries Kubernetes endpoints via control plane
-endpoint to find `trustd` endpoints, and use `trustd` to issue the certficiate.
+Worker nodes wait for their `kubelet` to join the cluster.
```
Comment (smira) [inline website/content/docs/v0.15/Guides/troubleshooting-control-plane.md:477]: we should probably update the doc here, as with cluster discovery enabled (which is the default), worker nodes find the control plane nodes via the cluster discovery, so they can join on `apid` level before the `kubelet` is ready

File: website/content/docs/v0.15/Guides/troubleshooting-control-plane.md [hunk 2]
```diff
@@ -14,64 +14,94 @@ Kubernetes client configuration can be pulled from control plane nodes with `tal
...
+default of `443` is _required_.
+The `endpoint` above may be a DNS name or IP address, but it should be
+ultimately be directed to the _set_ of all controlplane nodes, as opposed to a
+single one.
+
+As mentioned above, this can be achieved by a number of strategies, including:
+
+- an external load balancer
+- DNS records
+- Talos-builtin shared IP ([VIP](../guides/vip/)
```
Comment (smira) [inline website/content/docs/v0.15/Guides/troubleshooting-control-plane.md:80]:
```suggestion
- Talos-builtin shared IP ([VIP](../vip/)
```

File: website/content/docs/v0.15/Guides/troubleshooting-control-plane.md [hunk 3]
```diff
@@ -14,64 +14,94 @@ Kubernetes client configuration can be pulled from control plane nodes with `tal
...
+
+Maintaining the proper count of control plane nodes is also critical.
+The etcd database operates on the principles of membership and quorum, so
+membership should always be an odd number, and there is exponentially-increasing
+overhead for each additional member.
+Therefore, the number of control plane nodes should almost always be 3.
+In some particularly large or distributed clusters, the count may be 5, but this
+is very rare.
+
+See [this document](../learn-more/concepts/#control-planes-are-not-linear-replicas) on the topic for more information.
```
Comment (smira) [inline website/content/docs/v0.15/Guides/troubleshooting-control-plane.md:51]:
```suggestion
See [this document](../../learn-more/concepts/#control-planes-are-not-linear-replicas) on the topic for more information.
```

**Review by smira — APPROVED** [review]

**Review by frezbo — APPROVED** [review]

File: website/content/docs/v0.15/Guides/troubleshooting-control-plane.md (hunk 1 above)
Comment (Ulexus) [reply website/content/docs/v0.15/Guides/troubleshooting-control-plane.md:477]: :+1: but probably needs more (and broader) work than this PR

**Pull request merged**